### DISCUSSION TITLE
Remove Images for which there are no annotations

### DISCUSSION THREAD

**Discussion by shanalikhan** [roboflow/supervision#1650] [Q&A] [answered]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Description

Hi, 
Any idea how to remove the images where there are no annotations or we need to do it manually in JSON in COCO Format?

### Use case

Cleaning COCO

### Additional

N/a

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by LinasKo** [comment]
Hi @shanalikhan,

I'm converting this to a discussion.

**Comment by shanalikhan** [comment]
I've written the manual code here:

```

coco_paths = [coco_output_train_folder,coco_output_test_folder]

### Removing the images with no annotations

for coco_path in coco_paths:
    coco_json = json.load(open(os.path.join(coco_path,'result.json')))
    print(f"Orignal Images Length : {len(coco_json['images'])}")
    missing_images = []
    for idx in tqdm(range(len(coco_json['images']))):
        img_id = coco_json['images'][idx]['id'] 
        img_annotations = list(filter(lambda x: x['image_id']==img_id,coco_json['annotations']))
        if len(img_annotations)==0:
            missing_images.append(idx)
    filtered_images = list(filter(lambda x: x['id'] not in missing_images,coco_json['images']))
    coco_json['images'] = filtered_images
    print(f"New Images Length : {len(filtered_images)}")
    json.dump(coco_json,open(os.path.join(coco_path,'result.json'),'w'))
            
```

**Reply by LinasKo** [reply, marked as answer]
Indeed, we do not have a quick way of doing it at the moment. I'd normally advise doing something similar to what you did.
